File: ruby/raindrops/raindrops.rb
# https://exercism.org/tracks/ruby/exercises/raindrops
class Raindrops
  def self.convert(number)
    result = ''
    factors = {
      3 => 'Pling',
      5 => 'Plang',
      7 => 'Plong'
    }

    factors.each do |factor, drop_sound|
      result += drop_sound if (number % factor).zero?
    end

    result.empty? ? number.to_s : result
  end
end

File: go/raindrops/raindrops.go
package raindrops

import (
	"fmt"
	"strings"
)

func Convert(number int) string {
	var result strings.Builder
	if number%3 == 0 {
		result.WriteString("Pling")
	}
	if number%5 == 0 {
		result.WriteString("Plang")
	}
	if number%7 == 0 {
		result.WriteString("Plong")
	}
	if result.String() == "" {
		return fmt.Sprint(number)
	}

	return result.String()
}
